City-of-Helsinki/drupal-helfi-platform-config
# UHF-10908 FunctionalJS test for cookie banner

# [UHF-10908](https://helsinkisolutionoffice.atlassian.net/browse/UHF-10908)


## What was done


* Use variable for the spacerParentSelector, if it is set.
* Added hdbt_cookie_banner_test module to alter settings in tests.
* Added test for the cookie banner functionality.

## How to install
* Make sure your instance is up and running on latest dev branch.
    * `git pull origin dev`
    * `make fresh`
* Update the Helfi Platform config
    * `composer require drupal/helfi_platform_config:dev-UHF-10908`
* Run `make drush-updb drush-cr`

## How to test


* Add `testing` profile to your instance `.env` file: 
   ```shell
   COMPOSE_PROFILES=testing
   ```
* Get the drush server up and running. In the container (make shell) run: 
   ```shell
   drush runserver https://app &
   ```
* Run the test
   ```shell
   vendor/bin/phpunit -c /app/phpunit.xml.dist public/modules/contrib/helfi_platform_config/modules/hdbt_cookie_banner/tests/src/FunctionalJavascript
   ```
* Check that code follows our standards


[UHF-10908]: https://helsinkisolutionoffice.atlassian.net/browse/UHF-10908?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
